commands/api/base.go

```
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/boltdb/bolt"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/TykTechnologies/tyk-cli/db"
+	"github.com/TykTechnologies/tyk-cli/utils"
+	"github.com/TykTechnologies/tyk/apidef"
+)
+
+type APIDef struct {
+	id            string
+	name          string
+	item          db.Item
+	APIModel      APIModel
+	APIDefinition apidef.APIDefinition
+}
+
+type APIModel struct {
+	schemaPath string
+}
+
+func (api *APIDef) setAPIDefinition() {
+	api.APIDefinition.APIID = api.Id()
+	api.APIDefinition.Name = api.Name()
+}
+
+func New(name string) *APIDef {
+	api := APIDef{}
+	api.id = strings.Replace(uuid.NewV4().String(), "-", "", -1)
+	api.name = name
+	return &api
+}
+
+func (api *APIDef) Id() string {
+	return api.id
+}
+
+func (api *APIDef) Name() string {
+	return api.name
+}
+
+func (api *APIDef) BucketName() string {
+	return "apis"
+}
+
+func (api *APIDef) Group() string {
+	return "API"
+}
+
+func (api *APIDef) GetRecordData() interface{} {
+	api.setAPIDefinition()
+	type rec struct {
+		APIModel      APIModel             `json:"api_model"`
+		APIDefinition apidef.APIDefinition `json:"api_definition"`
+	}
+	return rec{api.APIModel, api.APIDefinition}
+}
+
+// Create is a public function for creating staged APIs
+func (api *APIDef) Create(bdb *bolt.DB) error {
+	err := bdb.Update(func(tx *bolt.Tx) error {
+		return db.AddRecord(tx, api)
+	})
+	return utils.ReturnErr(err)
+}
+
+// Find is a public function for finding staged APIs
+func (apis *APIDef) Find(bdb *bolt.DB, id string) (interface{}, error) {
+	var item interface{}
+	err := bdb.View(func(tx *bolt.Tx) error {
+		collection := tx.Bucket([]byte(apis.BucketName()))
+		member := collection.Get([]byte(id))
+		err := json.Unmarshal(member, &item)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if item == nil {
+		log.Fatal("API not found")
```

If you want to return a better error, do something like `return fmt.Errorf("API not found: %v", err)`